### ISSUE TITLE
Hi NoodleOfDeath

### ISSUE THREAD

**Issue by gogolith** [NoodleOfDeath/NoodleKit#1]
Sorry, i didn't found another way to contact you. You have written an iOS review about my Code Editor right? If yes please contact me develop@gogolith.de

Thank you.

regards 
Christoph

**Comment by NoodleOfDeath** [comment]
That's completely okay! I'm happy you reached out to me! I'm actually 
working on a code editor myself. A free version hopefully, so we may be 
rivals, but I'm happy to meet a fellow programmer!

Best,


Randolph-Macon College Logo <http://www.rmc.edu/>


  Thom Morgan

Mobile: (703) 678-5982
www.muxcode.com/byteme/ <http://www.muxcode.com/byteme/>
Randolph-Macon College 2016
Computer Science & Mathematics Major
Film Studies Minor
ByteMe © Code Editor | Founder & Developer
Chi Beta Phi Honorary Fraternity | National Historian | Historian (R-MC)
Phi Kappa Sigma, Tau Chapter | Treasurer & Philanthropy Chair (R-MC)
Habitat For Humanity | Treasurer (R-MC)
ARCADE Gaming Society | Founder & Advisor (R-MC)
IRIS Film & Photography Club | Founder & Advisor (R-MC)
Roomance © Non-Profit Advocacy Organization | Founder & Developer

**Comment by gogolith** [comment]
nice happy to be right here... you wrote in your review something about a saving bug (wrong content). Can you tell me more about that issue.

**Comment by NoodleOfDeath** [comment]
Yes, I found that if you are editing a syntax mode and switching between 
a file that implements that syntax mode, and a regular expression error 
faults, the loading of the file gets dicey and sometimes overwrites one 
of the two open files with the contents of the other. Not sure why that 
would happen unless your editors do not do saves to individually 
separate documents. I don't know your implementation but it would seem 
to me that your workspace uses a single document as the focused document 
and all write/reads are made to that document meaning that when 
switching rapidly between files during write/read processes could lead 
to some unexpected, unwanted results?

Best,

Thom

On 5/25/16 10:34 AM, Christoph Gogolin wrote:
>
> nice happy to be right here... you wrote in your review something 
> about a saving bug (wrong content). Can you tell me more about that 
> issue.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub 
> <https://github.com/NoodleOfDeath/NoodleKit/issues/1#issuecomment-221595691>
>

-- 
Randolph-Macon College Logo <http://www.rmc.edu/>


  Thom Morgan

Mobile: (703) 678-5982
www.muxcode.com/byteme/ <http://www.muxcode.com/byteme/>
Randolph-Macon College 2016
Computer Science & Mathematics Major
Film Studies Minor
ByteMe © Code Editor | Founder & Developer
Chi Beta Phi Honorary Fraternity | National Historian | Historian (R-MC)
Phi Kappa Sigma, Tau Chapter | Treasurer & Philanthropy Chair (R-MC)
Habitat For Humanity | Treasurer (R-MC)
ARCADE Gaming Society | Founder & Advisor (R-MC)
IRIS Film & Photography Club | Founder & Advisor (R-MC)
Roomance © Non-Profit Advocacy Organization | Founder & Developer

**Comment by gogolith** [comment]
where was your file with your syntax mode located? local or on a remote?

**Comment by NoodleOfDeath** [comment]
local


On 5/25/16 10:41 AM, Christoph Gogolin wrote:
>
> where was your file with your syntax mode located? local or on a remote?
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub 
> <https://github.com/NoodleOfDeath/NoodleKit/issues/1#issuecomment-221597868>
>

-- 
Randolph-Macon College Logo <http://www.rmc.edu/>


  Thom Morgan

Mobile: (703) 678-5982
www.muxcode.com/byteme/ <http://www.muxcode.com/byteme/>
Randolph-Macon College 2016
Computer Science & Mathematics Major
Film Studies Minor
ByteMe © Code Editor | Founder & Developer
Chi Beta Phi Honorary Fraternity | National Historian | Historian (R-MC)
Phi Kappa Sigma, Tau Chapter | Treasurer & Philanthropy Chair (R-MC)
Habitat For Humanity | Treasurer (R-MC)
ARCADE Gaming Society | Founder & Advisor (R-MC)
IRIS Film & Photography Club | Founder & Advisor (R-MC)
Roomance © Non-Profit Advocacy Organization | Founder & Developer

**Comment by gogolith** [comment]
Ok thanks that should help me to reproduce it. Can you also tell me something about your setup? iOS version and Device? External keyboard?

**Comment by NoodleOfDeath** [comment]
iPhone 6 no external keyboard and iPad Air 2 no external keyboard. Do 
you have a beta software program? I would be more than willing to help out!

Best,

On 5/25/16 10:45 AM, Christoph Gogolin wrote:
>
> Ok thanks that should help me to reproduce it. Can you also tell me 
> something about your setup? iOS version and Device? External keyboard?
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub 
> <https://github.com/NoodleOfDeath/NoodleKit/issues/1#issuecomment-221599024>
>

-- 
Randolph-Macon College Logo <http://www.rmc.edu/>


  Thom Morgan

Mobile: (703) 678-5982
www.muxcode.com/byteme/ <http://www.muxcode.com/byteme/>
Randolph-Macon College 2016
Computer Science & Mathematics Major
Film Studies Minor
ByteMe © Code Editor | Founder & Developer
Chi Beta Phi Honorary Fraternity | National Historian | Historian (R-MC)
Phi Kappa Sigma, Tau Chapter | Treasurer & Philanthropy Chair (R-MC)
Habitat For Humanity | Treasurer (R-MC)
ARCADE Gaming Society | Founder & Advisor (R-MC)
IRIS Film & Photography Club | Founder & Advisor (R-MC)
Roomance © Non-Profit Advocacy Organization | Founder & Developer

**Comment by gogolith** [comment]
Yes of curse. Please tell me you mail address. I will add you to the next beta. 
Thank you. 
I will also test your regex issue.

**Comment by NoodleOfDeath** [comment]
Awesome sauce! My email is thommy.morgan@gmail.com

Best,

On 5/25/16 10:50 AM, Christoph Gogolin wrote:
>
> Yes of curse. Please tell me you mail address. I will add you to the 
> next beta.
> Thank you.
> I will also test your regex issue.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub 
> <https://github.com/NoodleOfDeath/NoodleKit/issues/1#issuecomment-221600614>
>

-- 
Randolph-Macon College Logo <http://www.rmc.edu/>


  Thom Morgan

Mobile: (703) 678-5982
www.muxcode.com/byteme/ <http://www.muxcode.com/byteme/>
Randolph-Macon College 2016
Computer Science & Mathematics Major
Film Studies Minor
ByteMe © Code Editor | Founder & Developer
Chi Beta Phi Honorary Fraternity | National Historian | Historian (R-MC)
Phi Kappa Sigma, Tau Chapter | Treasurer & Philanthropy Chair (R-MC)
Habitat For Humanity | Treasurer (R-MC)
ARCADE Gaming Society | Founder & Advisor (R-MC)
IRIS Film & Photography Club | Founder & Advisor (R-MC)
Roomance © Non-Profit Advocacy Organization | Founder & Developer

**Comment by gogolith** [comment]
If you have a beta from your editor, please add me too. :) 
develop@gogolith.de

**Comment by gogolith** [comment]
lets talk about the regex issue... 
i tested a bit... and .* works well.

**Comment by NoodleOfDeath** [comment]
Will do! Not at that step just yet!


On 5/25/16 10:54 AM, Christoph Gogolin wrote:
>
> If you have a beta from your editor, please add me too. :)
> develop@gogolith.de <mailto:develop@gogolith.de>
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub 
> <https://github.com/NoodleOfDeath/NoodleKit/issues/1#issuecomment-221601733>
>

-- 
Randolph-Macon College Logo <http://www.rmc.edu/>


  Thom Morgan

Mobile: (703) 678-5982
www.muxcode.com/byteme/ <http://www.muxcode.com/byteme/>
Randolph-Macon College 2016
Computer Science & Mathematics Major
Film Studies Minor
ByteMe © Code Editor | Founder & Developer
Chi Beta Phi Honorary Fraternity | National Historian | Historian (R-MC)
Phi Kappa Sigma, Tau Chapter | Treasurer & Philanthropy Chair (R-MC)
Habitat For Humanity | Treasurer (R-MC)
ARCADE Gaming Society | Founder & Advisor (R-MC)
IRIS Film & Photography Club | Founder & Advisor (R-MC)
Roomance © Non-Profit Advocacy Organization | Founder & Developer

**Comment by NoodleOfDeath** [comment]
Then perhaps it's just Swift :D


On 5/25/16 11:03 AM, Christoph Gogolin wrote:
>
> lets talk about the regex issue...
> i tested a bit... and .* works well.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub 
> <https://github.com/NoodleOfDeath/NoodleKit/issues/1#issuecomment-221604552>
>

-- 
Randolph-Macon College Logo <http://www.rmc.edu/>


  Thom Morgan

Mobile: (703) 678-5982
www.muxcode.com/byteme/ <http://www.muxcode.com/byteme/>
Randolph-Macon College 2016
Computer Science & Mathematics Major
Film Studies Minor
ByteMe © Code Editor | Founder & Developer
Chi Beta Phi Honorary Fraternity | National Historian | Historian (R-MC)
Phi Kappa Sigma, Tau Chapter | Treasurer & Philanthropy Chair (R-MC)
Habitat For Humanity | Treasurer (R-MC)
ARCADE Gaming Society | Founder & Advisor (R-MC)
IRIS Film & Photography Club | Founder & Advisor (R-MC)
Roomance © Non-Profit Advocacy Organization | Founder & Developer

**Comment by gogolith** [comment]
;)
I think i found something that can throw an error on loading an invalide mode file... e.g. you switch to a state that doesn't exists can make the editor unstable. That could be the reason for the saving issues too. 

Thank you very much for that. 
I will fix it with the next release and hope for a better review... :) 

(Delete backwards is also already fixed) 


regards

**Comment by NoodleOfDeath** [comment]
Perfect! :D


On 5/25/16 11:25 AM, Christoph Gogolin wrote:
>
> ;)
> I think i found something that can throw an error on loading an 
> invalide mode file... e.g. you switch to a state that doesn't exists 
> can make the editor unstable. That could be the reason for the saving 
> issues too.
>
> Thank you very much for that.
> I will fix it with the next release and hope for a better review... :)
>
> (Delete backwards is also already fixed)
>
> regards
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub 
> <https://github.com/NoodleOfDeath/NoodleKit/issues/1#issuecomment-221610997>
>

-- 
Randolph-Macon College Logo <http://www.rmc.edu/>


  Thom Morgan

Mobile: (703) 678-5982
www.muxcode.com/byteme/ <http://www.muxcode.com/byteme/>
Randolph-Macon College 2016
Computer Science & Mathematics Major
Film Studies Minor
ByteMe © Code Editor | Founder & Developer
Chi Beta Phi Honorary Fraternity | National Historian | Historian (R-MC)
Phi Kappa Sigma, Tau Chapter | Treasurer & Philanthropy Chair (R-MC)
Habitat For Humanity | Treasurer (R-MC)
ARCADE Gaming Society | Founder & Advisor (R-MC)
IRIS Film & Photography Club | Founder & Advisor (R-MC)
Roomance © Non-Profit Advocacy Organization | Founder & Developer

**Comment by gogolith** [comment]
I also seen in your example is one thing wrong. 
you double escaped quotes 
"regex" : "\\"" should be -> "regex" : "\""
same here "regex" : "[^\"]*\\""  -> "regex" : "[^\"]*\""

**Comment by NoodleOfDeath** [comment]
I escaped the double quotes because you can escape " in regex. Just a habit.

Best,


On 5/25/16 11:44 AM, Christoph Gogolin wrote:
>
> I also seen in your example is one thing wrong.
> you double escaped quotes
> "regex" : "\\"" should be -> "regex" : "\""
> same here "regex" : "[^\"]/\\"" -> "regex" : "[^\"]/\""
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub 
> <https://github.com/NoodleOfDeath/NoodleKit/issues/1#issuecomment-221614972>
>

-- 
Randolph-Macon College Logo <http://www.rmc.edu/>


  Thom Morgan

Mobile: (703) 678-5982
www.muxcode.com/byteme/ <http://www.muxcode.com/byteme/>
Randolph-Macon College 2016
Computer Science & Mathematics Major
Film Studies Minor
ByteMe © Code Editor | Founder & Developer
Chi Beta Phi Honorary Fraternity | National Historian | Historian (R-MC)
Phi Kappa Sigma, Tau Chapter | Treasurer & Philanthropy Chair (R-MC)
Habitat For Humanity | Treasurer (R-MC)
ARCADE Gaming Society | Founder & Advisor (R-MC)
IRIS Film & Photography Club | Founder & Advisor (R-MC)
Roomance © Non-Profit Advocacy Organization | Founder & Developer
